### PR TITLE
DT-961 Save, delete and exists index methods

### PR DESCRIPTION
**Pull request by andymarke**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/indexer/repository/OffenderRepositoryIntegrationTest.kt [hunk 1]
```diff
@@ -78,91 +83,225 @@ internal class OffenderRepositoryIntegrationTest : IntegrationTest() {
...
+        offenderRepository.createIndex(GREEN)
+
+        val response = highLevelClient.ingest().getPipeline(GetPipelineRequest("pnc-pipeline"), RequestOptions.DEFAULT)
+        assertThat(response.isFound).isTrue()
+      }
+    }
+
   }
 
   @Nested
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/indexer/repository/OffenderRepositoryIntegrationTest.kt:179]: Changed the nesting so changes look bigger than they are

File: src/test/kotlin/uk/gov/justice/digital/hmpps/indexer/service/CommunityServiceTest.kt [hunk 2]
```diff
@@ -18,7 +18,7 @@ internal class CommunityServiceTest : IntegrationTest() {
     CommunityApiExtension.communityApi.stubFor(WireMock.get(WireMock.anyUrl()).willReturn(WireMock.aResponse()
         .withHeader("Content-Type", "application/json")
         .withBody("""{
-          "offenderId": 99
+          "offenderId": 99,
```
Comment (andymarke) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/indexer/service/CommunityServiceTest.kt:21]: oops

File: src/test/kotlin/uk/gov/justice/digital/hmpps/indexer/repository/OffenderRepositoryIntegrationTest.kt [hunk 3]
```diff
@@ -78,91 +83,225 @@ internal class OffenderRepositoryIntegrationTest : IntegrationTest() {
...
     @Test
-    internal fun `dateOfBirth is a date`() {
-      val value = mappingProperties.value("dateOfBirth.type")
-      assertThat(value).isEqualTo("date")
+    internal fun `will save json`() {
+      offenderRepository.save(Offender(OffenderDetail(crn = "X12345", offenderId = 99).asJson()), BLUE)
+
+      val json = highLevelClient.get(GetRequest(BLUE.indexName).id("X12345"), RequestOptions.DEFAULT).sourceAsString
+
+      assertThatJson(json).isEqualTo("{\"offenderId\":99,\"crn\":\"X12345\"}")
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/indexer/repository/OffenderRepositoryIntegrationTest.kt:203]: Is this a job for the three quotes """?

Comment (andymarke) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/indexer/repository/OffenderRepositoryIntegrationTest.kt:203]: yes, would be a bit more readable

**Pull request merged**